src/stdlib.rs

```diff
@@ -62,7 +62,17 @@ impl<F: Fn(Tree) + Clone + Send + Sync + 'static> AsHostedDef for LogDef<F> {
 #[cfg(feature = "std")]
 #[allow(clippy::absolute_paths)]
 pub fn create_host(book: &crate::ast::Book) -> Arc<Mutex<Host>> {
-  let host = Arc::new(Mutex::new(Host::default()));
+  let host: Arc<Mutex<Host>> = Default::default();
+
+  insert_book(host.clone(), book);
+
+  host
+}
+
+/// Create a `Host` from a `Book`, including `hvm-core`'s built-in definitions
+#[cfg(feature = "std")]
+#[allow(clippy::absolute_paths)]
+pub fn insert_book(host: Arc<Mutex<Host>>, book: &crate::ast::Book) {
```

Comment:
IMO this should be `insert_host` and just insert the IO defs, without touching the book